### PR TITLE
[DependencyInjection] Add `#[AutowireCallable]` and `#[AutowireServiceClosure]`

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Fixes https://github.com/symfony/symfony-docs/issues/18034

### CODE REVIEW

File: service_container/service_closures.rst [hunk 1]
```diff
@@ -92,6 +92,49 @@ argument of type ``service_closure``:
...
+    class MyService
+    {
+        /**
+         * @var callable(): MailerInterface
+         */
+        private \Closure $mailer;
+
+        public function __construct(
+            #[AutowireServiceClosure('mailer')]
+            \Closure $mailer
```
Comment (OskarStark) [inline service_container/service_closures.rst]: CPP?

File: service_container.rst [hunk 2]
```diff
@@ -796,6 +798,54 @@ Our configuration looks like this:
...
+    use Psr\Log\LoggerInterface;
+    use Symfony\Component\DependencyInjection\Attribute\AutowireCallable;
+
+    class MessageGenerator
+    {
+        private $logger;
+        private $messageHash;
+
+        public function __construct(
+            LoggerInterface $logger,
```
Comment (OskarStark) [inline service_container.rst]: CPP?

File: service_container/service_closures.rst (hunk 1 above)
Comment (alexandre-daubois) [reply service_container/service_closures.rst]: Ah yes!

File: service_container.rst (hunk 2 above)
Comment (alexandre-daubois) [reply service_container.rst]: Updated as well 🙂

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
Thank you Alexandre.

**Pull request merged**